### PR TITLE
fix the error caused by helm lint not pulling values outside of the default value for schema validation

### LINKED ISSUE #8880: Helm lint not pulling values outside of the default values file for schema validation
Hi 

Im not 100% if this is a bug or the intended functionality of the helm v3 lint command but I noticed an issue when I uplifted our ci pipeline from helm v2 to helm v3 (see actual versions below)

What is currently happening?
when I run `helm lint . --values=second_values.yaml` any value that is in second_values.yaml that is also in the values.yaml file of the chart will not be overridden with the values of second_values.yaml. 
Example
image

I was able to recreate the issue with a test chart (also attached as [testchart-0.1.0.zip](https://github.com/helm/helm/files/5369833/testchart-0.1.0.zip))

I dont know if this is a change in functionality. I am aware that the helm lint command handles schema checks differently between helm v2 and helm v3 however the schema used to validate the helm v3 lint command should be correct provided the second_values.yaml file is used.

Any help on this issue/question would be helpful as the only way we can work around this is by either disabling the helm lint command in our ci pipelines or removing some validation from our values.schema.json file

Output of `helm version`:
helm 3 installation
`helm version
version.BuildInfo{Version:"v3.3.4", GitCommit:"a61ce5633af99708171414353ed49547cf05013d", GitTreeState:"clean", GoVersion:"go1.14.9"}
`
helm 2 installation
`
helm2 version
Client: &version.Version{SemVer:"v2.16.5", GitCommit:"89bd14c1541fa93a09492010030fd3699ca65a97", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.1", GitCommit:"65cac393eeadea00384ea7108296ee6e5509c4de", GitTreeState:"clean"}
`

Output of `kubectl version`:
`kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:49Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-16T17:47:13Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}`

Cloud Provider/Platform (AKS, GKE, Minikube etc.):  N/A

### PR DESCRIPTION
**Pull request by donggangcj**
**What this PR does / why we need it**:

Close #8880

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [*] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by hickeyma — COMMENTED** [review]
@donggangcj Can you check out the failed test in CI?

**Comment by donggangcj** [comment]
> @donggangcj Can you check out the failed test in CI?

Yeap! I am solving this error.

File: pkg/lint/rules/values.go [hunk 1]
```diff
@@ -60,6 +60,14 @@ func validateValuesFileExistence(valuesPath string) error {
 	return nil
 }
 
+func copyMap(src map[string]interface{}) map[string]interface{} {
```
Comment (donggangcj) [inline pkg/lint/rules/values.go:63]: @hickeyma  Does `helm`  have a method for map copy?

**Comment by donggangcj** [comment]
@hickeyma Done;

> @donggangcj Can you check out the failed test in CI?

Done;

**Review by bacongobbler — CHANGES_REQUESTED** [review]
Hi @donggangcj. As requested in previous PRs, we ask that submissions present a unit test to ensure that the fix resolves the issue. Please write a unit test that demonstrates the issue as well as why the fix presented here resolves the issue.

Thanks.

File: pkg/lint/rules/values.go (hunk 1 above)
Comment (bacongobbler) [reply pkg/lint/rules/values.go:63]: No.

**Comment by mluckam** [comment]
Hello, I have also ran into this issue and would like to see it resolved.  @bacongobbler I created the requested unit test along with necessary code changes the unit test exposed, but I do not have access to push to this pull request.  How should I proceed?

**Pull request closed without merging**